packages/common/test/index.test.js

```diff
@@ -866,95 +867,172 @@ describe('validate', () => {
-    const state = {}
-    const c = { page: 22, next: 23 }
-    const result = cursor(c, { key: 'cursor' })(state)
-    expect(result.cursor).to.eql(c)
+    const state = {};
+    const c = { page: 22, next: 23 };
+    const result = cursor(c, { key: 'cursor' })(state);
+    expect(result.cursor).to.eql(c);
   });
 
   it('should apply a custom formatter', () => {
     const state = {};
     const result = cursor('abc', {
-      format: (c) => c.toUpperCase()
+      format: c => c.toUpperCase(),
     })(state);
-    expect(result.cursor).to.eql('ABC')
-  })
+    expect(result.cursor).to.eql('ABC');
+  });
 
   it('should format "today"', () => {
     const state = {};
     const date = new Date().toDateString();
     const result = cursor('today', {
-      format: (c) => c.toDateString()
+      format: c => c.toDateString(),
     })(state);
-    expect(result.cursor).to.eql(date)
-  })
+    expect(result.cursor).to.eql(date);
+  });
 
   it('should format a number to an arbitrary object', () => {
     const state = {};
     const result = cursor(3, {
-      format: (c) => ({ page: c })
+      format: c => ({ page: c }),
     })(state);
-    expect(result.cursor).to.eql({ page: 3 })
-  })
+    expect(result.cursor).to.eql({ page: 3 });
+  });
+});
+
+describe('group', () => {
+  it('should group an array of objects by a specified key path', function () {
+    const state = {};
+    const data = [{ x: 'a' }, { x: 'b' }, { x: 'b' }];
+
+    const result = group(data, 'x')(state);
+
+    expect(result.data).eql({ a: [{ x: 'a' }], b: [{ x: 'b' }, { x: 'b' }] });
+  });
 
-});
\ No newline at end of file
+  it('should group an array of objects by a specified path', function () {
+    const data = [
+      { x: 'a', y: { z: 'a' } },
+      { x: 'b', y: { z: 'a' } },
+    ];
+    const state = {};
+    const result = group(data, 'y.z')(state);
+    expect(result.data).eql({
+      a: [
+        { x: 'a', y: { z: 'a' } },
+        { x: 'b', y: { z: 'a' } },
+      ],
+    });
+  });
+  it("should return an empty object if the key isn't present on any items", function () {
+    const data = [
+      { x: 'a', y: { z: 'a' } },
+      { x: 'b', y: { w: 'a' } },
+    ];
+    const state = {};
+    const result = group(data, 'y.q')(state);
+    expect(result.data).eql({});
+  });
+
+  it('should remove undefined keys', function () {
+    const data = [
+      { x: 'a', y: { z: 'a' } },
+      { x: 'b', y: { z: undefined } },
+    ];
+    const state = {};
+    const result = group(data, 'y.z')(state);
+    expect(result.data).eql({ a: [{ x: 'a', y: { z: 'a' } }] });
+  });
+  it('should expand arrayOfObjects', function () {
+    const input = {
+      data: [
+        { x: 'a', y: { z: 'a' } },
+        { x: 'b', y: { z: undefined } },
+      ],
+    };
+
+    let result;
+    group(state => {
+      result = state.data;
+      return result;
+    }, 'y.z')(input);
+
+    expect(result).eql(input.data);
+  });
+
+  it('should expand key path', function () {
```

Comment:
Again, this one is also testing the arguments to the expanded function and that's not really the point. Let's just test the overall result with `group(state.data, () => 'x')`